react/AppStorePrice.tsx

```diff
@@ -0,0 +1,166 @@
+  },
+})
+
+const ListItem: FunctionComponent<{ paddingTop?: boolean }> = ({
+  children,
+  paddingTop,
+}) => {
+  const handles = useCssHandles(CSS_HANDLES)
+  return (
+    <div
+      className={`${handles.listItem} flex bb b--muted-4 pb6 ${
+        paddingTop ? 'pt6' : ''
+      }`}
+    >
+      {children}
+    </div>
+  )
+}
+
+const ItemList: FunctionComponent<{
+  description: string
+  formattedValue: string
+}> = ({ description, formattedValue }) => {
+  const handles = useCssHandles(CSS_HANDLES)
+  return (
+    <Fragment>
+      <span className={`${handles.itemListFirst} mid-gray w-60`}>
+        {description}
+      </span>
+      <span className={`${handles.itemListSecond} mid-gray w-40 f5`}>
+        {formattedValue}
+      </span>
+    </Fragment>
+  )
+}
+
+const RangeItem: FunctionComponent<{
+  value: number
+  metricName: string
+  minQuantity?: number
+  maxQuantity?: number
+}> = ({ value, metricName, minQuantity, maxQuantity }) => {
+  const metricPrice = useFormattedPrice(value ?? 0)
+  const intl = useIntl()
+
+  let metricDescription = ''
+
+  if (minQuantity && maxQuantity) {
+    metricDescription = intl.formatMessage(messages.metricLabel, {
+      minQuantity,
+      maxQuantity,
+    })
+  } else if (maxQuantity) {
+    metricDescription = intl.formatMessage(messages.metricLabelUpTo, {
+      maxQuantity,
+    })
+  } else {
+    metricDescription = intl.formatMessage(messages.metricLabelOver, {
+      minQuantity,
+    })
+  }
+  return (
+    <ListItem paddingTop>
+      <ItemList
+        description={`${metricDescription} ${metricName}`}
+        formattedValue={`${metricPrice} / ${metricName}`}
+      />
+    </ListItem>
+  )
+}
+
+const MetricItem: FunctionComponent<{ metric: Metric }> = ({ metric }) => {
+  const numberOfRanges = metric.ranges.length
+  return (
+    <Fragment>
+      {metric.ranges.map((metricRange, index) => (
+        <RangeItem
+          key={index}
+          metricName={metric.name}
+          value={metricRange.multiplier}
+          minQuantity={index === 0 ? undefined : metricRange.minQuantity}
+          maxQuantity={
+            index + 1 === numberOfRanges
+              ? undefined
+              : metric.ranges[index + 1].minQuantity
+          }
+        />
+      ))}
+    </Fragment>
+  )
+}
+
+const Subscription: FunctionComponent<{ subscription?: number | null }> = ({
+  subscription,
+}) => {
+  const subscriptionPrice = useFormattedPrice(subscription ?? 0)
+  const intl = useIntl()
+  return (
+    <ListItem>
+      <ItemList
+        description={intl.formatMessage(messages.subscriptionLabel)}
+        formattedValue={subscriptionPrice}
+      />
+    </ListItem>
+  )
+}
+
+interface AppPriceQuery {
+  appPrice: AppPrice
+}
+
+const AppStorePrice: FunctionComponent = () => {
+  const {
+    product: { productId },
+  } = useProduct()
+
+  const handles = useCssHandles(CSS_HANDLES)
+
+  const { data, loading } = useQuery<AppPriceQuery>(APP_PRICE, {
```

Comment:
You may want to use the types generated by `vtex setup` to variables as well
```suggestion
  const { data, loading } = useQuery<AppPriceQuery, AppPriceQueryVariables>(APP_PRICE, {
```